packages/material-ui/src/SnackbarContent/SnackbarContent.test.js

```diff
@@ -31,34 +31,38 @@ describe('<SnackbarContent />', () => {
   describe('prop: action', () => {
     it('should render the action', () => {
       const action = <span>action</span>;
-      const wrapper = shallow(<SnackbarContent message="message" action={action} />);
-      assert.strictEqual(wrapper.childAt(1).hasClass(classes.action), true);
-      assert.strictEqual(wrapper.childAt(1).contains(action), true);
+      const { container, getByText } = render(
+        <SnackbarContent message="message" action={action} />,
+      );
+      expect(container.querySelector(`.${classes.action}`)).to.have.class(classes.action);
+      expect(getByText('action')).to.have.property('nodeName', 'SPAN');
     });
 
     it('should render an array of elements', () => {
       const action0 = <span key={0}>action0</span>;
       const action1 = <span key={1}>action1</span>;
-      const wrapper = shallow(<SnackbarContent message="message" action={[action0, action1]} />);
-      assert.strictEqual(wrapper.childAt(1).hasClass(classes.action), true);
-      assert.strictEqual(wrapper.childAt(1).contains(action0), true);
-      assert.strictEqual(wrapper.childAt(1).contains(action1), true);
+      const { getByText } = render(
+        <SnackbarContent message="message" action={[action0, action1]} />,
+      );
+      expect(getByText('action0')).to.not.be.null;
+      expect(getByText('action1')).to.not.be.null;
     });
   });
 
   describe('prop: message', () => {
     it('should render the message', () => {
-      const message = <span>message</span>;
-      const wrapper = shallow(<SnackbarContent message={message} />);
-      assert.strictEqual(wrapper.childAt(0).hasClass(classes.message), true);
-      assert.strictEqual(wrapper.childAt(0).contains(message), true);
+      const message = 'message prop text';
+      const { getAllByRole } = render(<SnackbarContent message={<span>{message}</span>} />);
+      expect(getAllByRole('alert')[1]).to.have.text(message);
     });
   });
 
   describe('prop: role', () => {
     it('should render the role', () => {
```

Comment:
```suggestion
    it('can override the role', () => {
```

and then we should add a test for the default role. You could duplicate this test and then query for `alert` instead of `alertdialog`.

Reply:
Updated the description and added another test for the default role.